### ISSUE TITLE
Logistic Regression Issue

### ISSUE THREAD

**Issue by KeenCat** [autodeployai/pypmml#3]
Hello, Thanks for making this amazing library.

I tested logistic regression pmml file with iris dataset.

`from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
import pandas as pd
import numpy as np
#from sklearn_pmml_model.ensemble import PMMLForestClassifier

# Prepare data
iris = load_iris()
X = pd.DataFrame(iris.data)
X.columns = np.array(iris.feature_names)
y = pd.Series(np.array(iris.target_names)[iris.target])
y.name = "Class"
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=123)

from sklearn.linear_model import LogisticRegression

lrf = LogisticRegression().fit(X_train, y_train)
predicted = lrf.predict(X_test)
accuracy = accuracy_score(y_test, predicted)

from sklearn2pmml import sklearn2pmml
from sklearn2pmml import PMMLPipeline

pipeline = PMMLPipeline([("estimator", lrf)])
sklearn2pmml(pipeline, "Logistic.pmml")

from pypmml import Model

model = Model.fromFile('Logistic.pmml')
result = model.predict({'sepal_length': 5.1, 'sepal_width': 3.5, 'petal_length': 1.4, 'petal_width': 0.2})
#data = pd.read_csv('Iris.csv') 
#result = model.predict(data)
print(result)`

Here is my code.

It resulted nan prediction.
How can I solve this problem. Please help me. Thank you. 

![image](https://user-images.githubusercontent.com/8923887/63399026-2fab8e00-c40a-11e9-8175-335d5b67cf11.png)

**Comment by scorebot** [comment]
@KeenCat The issue is caused by the fields of input data NOT match ones defined in `Logistic.pmml`, you can open the PMML file, see the Data Dictionary :
```
	<DataDictionary>
		<DataField name="y" optype="categorical" dataType="string">
			<Value value="setosa"/>
			<Value value="versicolor"/>
			<Value value="virginica"/>
		</DataField>
		<DataField name="x1" optype="continuous" dataType="double"/>
		<DataField name="x2" optype="continuous" dataType="double"/>
		<DataField name="x3" optype="continuous" dataType="double"/>
		<DataField name="x4" optype="continuous" dataType="double"/>
	</DataDictionary>
```
The input fields of model are "x1", "x2", "x3", and "x4", not those "sepal_length", "sepal_width", "petal_length" and "petal_width", so you can get correct result using the following data:
```
result = model.predict({'x1': 5.1, 'x2': 3.5, 'x3': 1.4, 'x4': 0.2})
print(result)
```
The result likes:
```
{'probability(setosa)': 0.8377255619769726, 'probability(virginica)': 3.097085722417435e-05, 'probability(versicolor)': 0.1622434671658033}
```
I'm not familiar about sklearn2pmml how to use the actual names instead of "x1", "x2"..., "xN".

You could use the `nyoka` library to export PMML from sklearn, for example:
```
from sklearn.pipeline import Pipeline

pipeline_obj = Pipeline([
    ('estimator', LogisticRegression())
])
pipeline_obj.fit(X_train, y_train)

from nyoka import skl_to_pmml
skl_to_pmml(pipeline_obj, iris.feature_names, "Class", "Logistic-new.pmml")
```
Then, predict the model by PyPMML:
```
model = Model.fromFile('Logistic-new.pmml')
result = model.predict({'sepal length (cm)': 5.1, 'sepal width (cm)': 3.5, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2})
print(result)
```

**Comment by KeenCat** [comment]
Wow, Thank you!! I could have checked pmml file first. Thank you very much!

**Comment by KeenCat** [comment]
Okay, I would use nyoka library! Thank you for advise. PyPmml helps us very much. Thank you for making this library. It is hard to find the library to import pmml file into python. I will close the issue.

**Comment by scorebot** [comment]
You are welcome. Besides of PMML content, You can use those property functions of PyPMML Model to get model info, for example:
```
print(model.inputNames)
```
The result:
```
['x1', 'x2', 'x3', 'x4']
```
More info about the functions, you can see the source code.

**Comment by KeenCat** [comment]
However, I tested Random Forest model. In pmml file, there are no valuable names, either. But it works properly. What is the difference in this?
<pre><code>
# Random Forest
rf = RandomForestClassifier(n_estimators=100, oob_score=True, random_state = 123456)
rf.fit(X_train, y_train)

from sklearn2pmml import sklearn2pmml
from sklearn2pmml import PMMLPipeline


pipeline = PMMLPipeline([("estimator", rf)])
sklearn2pmml(pipeline, "RandomForest.pmml")

from pypmml import Model

model = Model.fromFile('RandomForest.pmml')
result = model.predict({'sepal_length': 5.1, 'sepal_width': 3.5, 'petal_length': 1.4, 'petal_width': 0.2})
print(result)
</code></pre>

![image](https://user-images.githubusercontent.com/8923887/63404368-e5340c80-c41d-11e9-8b96-02b07087e726.png)
![image](https://user-images.githubusercontent.com/8923887/63404379-f250fb80-c41d-11e9-89f0-5c87c712abce.png)

**Comment by scorebot** [comment]
The results may be not correct, for such case, the PMML model takes all input fields as missing. Some model types can handle missing values, that means can output valid values instead of null when there are missing inputs, like the tree model. Some model types can not handle missing values, any missing field could lead to the final null result, like regression model.

**Comment by KeenCat** [comment]
!! Okay, I understood! That's the reason the model predicts all data to only one result. (Example, in iris dataset, all the data is converged to virginica!) Thank you for answering me! and sorry for bothering you with simple question.
